addons/hr_expense/models/hr_expense_sheet.py

```diff
@@ -464,6 +464,22 @@ def create(self, vals_list):
         sheets.activity_update()
         return sheets
 
+    def write(self, values):
+        """ Ensures there is no empty expense report in a state different from draft or cancel """
+        if 'state' in values and values['state'] or 'expense_line_ids' in values or 'approval_state' in values:
+            for sheet in self:
+                state = values.get('state', sheet.state)
+                if values.get('expense_line_ids', sheet.expense_line_ids):
```

Comment:
I that `values.get('expense_line_ids')` will most probably be an ORM command and thus always be Truthy.

Reply:
:man_facepalming: